components/reference-implementation-openbanking-consumerdatastandards-au/src/main/java/org/wso2/openbanking/consumerdatastandards/au/extensions/exceptions/CDSConsentException.java

```diff
@@ -0,0 +1,199 @@
+        return errorEnum.getCode();
+    }
+
+    /**
+     * Get the error title.
+     *
+     * @return error title
+     */
+    public String getErrorTitle() {
+        return errorEnum.getTitle();
+    }
+
+    /**
+     * Get the custom detail if provided, otherwise the default detail.
+     *
+     * @return detail message
+     */
+    public String getErrorDetail() {
+        return customDetail != null ? customDetail : errorEnum.getDetail();
+    }
+
+    // Common factory method for any CDS error
+    /**
+     * Convert any Throwable to appropriate CDSConsentException based on exception type.
+     * This method maps different exception types to appropriate CDS errors.
+     *
+     * @param throwable the original exception
+     * @param context   optional context message for better error description
+     * @return CDSConsentException with appropriate CDS error enum
+     */
+    public static CDSConsentException fromThrowable(Throwable throwable, String context) {
+        String contextMessage = context != null ? context + ": " : "";
+
+        // If it's already a CDSConsentException, return as-is
+        if (throwable instanceof CDSConsentException) {
+            return (CDSConsentException) throwable;
+        }
+
+        // Map specific exception types to appropriate CDS errors
+        if (throwable instanceof JsonProcessingException) {
+            return new CDSConsentException(CDSErrorEnum.UNEXPECTED_ERROR,
+                    contextMessage + "JSON processing failed: " + throwable.getMessage(), throwable);
+        }
+
+        if (throwable instanceof NullPointerException) {
+            return new CDSConsentException(CDSErrorEnum.FIELD_MISSING,
+                    contextMessage + "Required field is missing or null", throwable);
+        }
+
+        if (throwable instanceof IllegalArgumentException) {
+            return new CDSConsentException(CDSErrorEnum.INVALID_FIELD,
+                    contextMessage + "Invalid field value: " + throwable.getMessage(), throwable);
+        }
+
+        if (throwable instanceof ClassCastException) {
+            return new CDSConsentException(CDSErrorEnum.INVALID_FIELD,
+                    contextMessage + "Invalid data type in request", throwable);
+        }
+
+        if (throwable instanceof NumberFormatException) {
+            return new CDSConsentException(CDSErrorEnum.INVALID_FIELD,
+                    contextMessage + "Invalid number format: " + throwable.getMessage(), throwable);
+        }
+
+        if (throwable instanceof SecurityException) {
+            return new CDSConsentException(CDSErrorEnum.UNAUTHORIZED,
+                    contextMessage + "Security validation failed", throwable);
+        }
+
+        // Default mapping for any other exception
+        return new CDSConsentException(CDSErrorEnum.UNEXPECTED_ERROR,
+                contextMessage + "An unexpected error occurred: " + throwable.getMessage(), throwable);
+    }
+}
```

Comment:
Add newline

Reply:
Fixed in [ab3cd77](https://github.com/wso2/reference-implementation-consumerdatastandards-au/pull/132/commits/ab3cd7712162cfa264437664bd0b42d28ff6034b)